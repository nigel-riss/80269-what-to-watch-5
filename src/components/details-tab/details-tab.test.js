import React from 'react';
import renderer from 'react-test-renderer';
import DetailsTab from './details-tab.jsx';


const filmMock = {
  cover: ``,
  poster: ``,
  preview: ``,
  name: ``,
  genre: `Biography`,
  year: 2004,
  description: ``,
  rating: 7.5,
  votes: 274,
  details: {
    director: `Martin Scorsese`,
    actors: [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Alec Baldwin`,
      `Kate Beckinsale`,
      `John C. Reilly`,
      `Alan Alda`,
      `Ian Holm`,
    ],
    length: {
      hours: 2,
      minutes: 50,
    },
  },
};


it(`Should render correctly`, () => {
  const tree = renderer
    .create(
        <DetailsTab
          film={filmMock}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
