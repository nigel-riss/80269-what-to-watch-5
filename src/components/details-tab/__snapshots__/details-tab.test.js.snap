// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Martin Scorsese
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Leonardo DiCaprio
        , 
        <br />
        Cate Blanchett
        , 
        <br />
        Alec Baldwin
        , 
        <br />
        Kate Beckinsale
        , 
        <br />
        John C. Reilly
        , 
        <br />
        Alan Alda
        , 
        <br />
        Ian Holm
        <br />
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        2
        h 
        50
        m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Biography
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2004
      </span>
    </p>
  </div>
</div>
`;
